tests/integration/test_valid_data_in_request.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: MIT
+
+import time
+
+import utils
+
+
+def test_valid_data_in_request(test_env):
+    """
+    Validate data in the request.
+
+    Process:
+    Send new request to the Cachito API
+    Send request to check status of existing request
+
+    Checks:
+    * Check that response code is 200
+    * Check in the response that state is complete
+    * Check that “packages” and “dependencies” keys have appropriate values
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response_created_req = client.create_new_request(
+        payload={
+            "repo": test_env["get"]["repo"],
+            "ref": test_env["get"]["ref"],
+            "pkg_managers": test_env["get"]["pkg_managers"],
+        },
+    )
+
+    response = client.fetch_request(response_created_req.id)
+    assert response.status == 200
+
+    start_time = time.time()
+    while response.data["state"] == "in_progress":
```

Comment:
Optional: It'd be nice to have this loop in a separate function so it can be reused for future tests.